### PR TITLE
add instructions for built-in TLS support

### PR DESCRIPTION
**Pull request by cmarcelk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -366,6 +365,48 @@ enable DEBUG logging in the SDK. You can add your own log statements in
...
+
+`Relay workflow server (relay-sdk-python/2.0.0) listening on 0.0.0.0 port 8080 with ssl_context MySslContext`
+
+When it says "ssl_context" instead of "plaintext", that is the clue that TLS is enabled.
+
+If you want to verify the certificate that your server is presenting to web clients, run:
+
+`openssl s_client -debug myserver.mydomain.com:8080`
+
+...assuming that you are still specifying port 8080. You may consider changing the port in
```
Comment (steveims) [inline README.md:403]: might be easier for the reader if we just present this with port == 443.
changing it seems like an advanced use case -- leave that to the adventurous, but no need to muddy this section?

File: README.md [hunk 2]
```diff
@@ -366,6 +365,48 @@ enable DEBUG logging in the SDK. You can add your own log statements in
...
+To use the built-in TLS support, the key and certificate both need to
+be in PEM format, and in a file that can be read locally by the workflow
+process.
+
+When not using TLS support, your workflow app should have a line that
+looks like this:
+
+`server = relay.workflow.Server('0.0.0.0', port)`
+
+When you want to use the built-in TLS support, you supply extra arguments
```
Comment (steveims) [inline README.md:385]: for Python, those "extra arguments" are "keyword arguments".

File: README.md (hunk 1 above)
Comment (cmarcelk) [reply README.md]: Does the 2nd commit address your point? I do agree it is an improvement.

File: README.md [hunk 3]
```diff
@@ -366,6 +365,41 @@ enable DEBUG logging in the SDK. You can add your own log statements in
...
+for providing TLS encryption, given that you have a TLS key and certificate.
+For example, you can get a TLS key and certificate from letsencrypt.org. Or
+you can use a commercial provider. Sometimes this key and certificate for
+HTTP-over-TLS is called an "SSL certificate".
+
+To use the built-in TLS support, the key and certificate both need to
+be in PEM format, and in a file that can be read locally by the workflow
+process.
+
+When not using TLS support, your workflow app should have a line that
```
Comment (steveims) [inline README.md:380]: Changes look good!

This comment about "not using TLS support" made me wonder ... have we given guidance anywhere that TLS is a "best practice"?  (Does ibot even require it?)

So "not using TLS support" in the SDK is fine when behind a proxy, etc with TLS support.

Comment (cmarcelk) [reply README.md:389]: This is why it's good to have extra eyes, that aren't contaminated with previous education ;-) See if the 3rd commit clarifies this sufficiently.

Comment (steveims) [reply README.md:389]: perfect!

**Pull request merged**